### ISSUE TITLE
How to reset Form?

### ISSUE THREAD

**Issue by jordan1982** [tkrotoff/react-form-with-constraints#22]
Hi,
how I can reset the form and clear all error?

Thank you

**Comment by tkrotoff** [comment]
Read this: [Clear and reset form input fields
](https://stackoverflow.com/questions/43922508)

**Comment by tkrotoff** [comment]
However the field feedbacks are not reset, I will fix this in the next version

**Comment by jordan1982** [comment]
ok, is there a way I can clean up field feedback?
Thank you

**Comment by jordan1982** [comment]
Waiting for the new version of the component,  to clean the error messages we used the updateFields method (calling it for each field).
It would be useful to implement a new method that cycles on the fields and clears all errors.
It's possible @tkrotoff ? Can I help you?


this.form.fieldsStore.updateField("name", {
      dirty: false,
       errors: new Set(),
       warnings: new Set(),
       infos: new Set(),
       validationMessage: ""
     });

Thank you

**Comment by tkrotoff** [comment]
> calling it for each field

No need, you can do this instead:
```TypeScript
  handleReset() {
    for (const fieldName in this.formWithConstraints.fieldsStore.fields) {
      this.formWithConstraints.fieldsStore.fields[fieldName] = {
        dirty: false,
        errors: new Set(),
        warnings: new Set(),
        infos: new Set(),
        validationMessage: ''
      };
    }
    this.formWithConstraints.forceUpdate();
  }
```

Don't forget also to re-initialize the state (`this.setState(this.getMyInitialState())`).

In the next version you will have `this.formWithConstraints.resetForm()`.

**Comment by jordan1982** [comment]
good job, tell me when new version is aivalable.
Thank you @tkrotoff

**Comment by tkrotoff** [comment]
I've released v0.8.0
There is now a `FormWithConstraints.reset()`, see https://github.com/tkrotoff/react-form-with-constraints/blob/v0.8.0/examples/Bootstrap4/App.jsx#L79

Took me quite some time, it's a full rewrite so a few things have changed due to Async support
